### PR TITLE
chore(helm platform): Add deprecation notice for rawConfig

### PR DESCRIPTION
**Pull request by spencergilbert**
As far as I can tell, the only notice of deprecation before was the comments in `values.yaml`.

This will provide a more official notice and let us remove `rawConfig` with the 0.15 release.

Signed-off-by: Spencer Gilbert <spencer.gilbert@gmail.com>

### CODE REVIEW

File: distribution/helm/vector-agent/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{{ .Chart.Name }} has been installed. Check it's status by running:
+  $ kubectl --namespace {{ .Release.Namespace}} get pods -l "app.kubernetes.io/instance={{ .Release.Name }}"
+
+WARNING: The `rawConfig` key has been deprecated and will be removed in an upcoming release.
```
Comment (jszwedko) [inline distribution/helm/vector-agent/templates/NOTES.txt:4]: Maybe we could add a link to the highlight that will be published? The naming is predictable, for example: `https://vector.dev/highlights/2021-04-21-vector-tap/`

File: docs/highlights/2021-06-01-removing-helm-rawconfig.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+   sinks:
+     stdout:
+       type: "console"
+       inputs: ["dummy"]
+-      rawConfig: |
+-        target = "stdout"
+-        encoding = "json"
++      target: "stdout"
++      encoding: "json"
+```
```
Comment (jszwedko) [inline docs/highlights/2021-06-01-removing-helm-rawconfig.md:42]: Could we add a link to our helm docs too for more details?

**Review by jszwedko — APPROVED** [review]
Just a couple of notes, but otherwise this looks great!

File: docs/highlights/2021-06-01-removing-helm-rawconfig.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+tags: ["type: announcement", "platform: kubernetes", "domain: config"]
+---
+
+With the release of Vector 0.14.0, we are announcing the planned deprecation of the `rawConfig` option
+in our Helm charts. This will remove a source of complexity in our chart logic, and is a step towards
+using configuration provided in the `values.yaml` without converting them for the `managed.toml` config file.
+
+## Upgrade Guide
+
+Below is an example of how to transition your `values.yaml` file:
```
Comment (StephenWakely) [inline docs/highlights/2021-06-01-removing-helm-rawconfig.md:19]: I think it is worth saying first something like,

> The values that were in rawConfig in toml format now need to be moved up a level and written as yaml.

**Review by StephenWakely — APPROVED** [review]

**Review by JeanMertz — APPROVED** [review]

**Pull request merged**